vowpalwabbit/interactions.h

```diff
@@ -398,7 +399,18 @@ struct interactions_generator
     for (auto ns_index : indices)
     {
       for (const auto& extent : feature_space[ns_index].namespace_extents)
-      { all_seen_extents.insert({ns_index, extent.hash}); }
+      {
+        // Interactions should not be generated for reserved namespaces such as
+        // constant. These reserved namespaces use their hash as the namespace
+        // character value so we can check if the value is in this range. There
+        // is a chance of collisions here though. 0 is a special case as the
+        // default case is mapped to a hash of 0 even though it is in slot ' ',
```

Comment:
super duper nit: maybe don't use `slot` here in the comment because slot is an overloaded term